release-notes.md

```diff
@@ -24,6 +33,85 @@
     + [Unit testing](#unit-testing)
     + [Integration tests parallelization](#integration-tests-parallelization)
 
+# v0.4.0
+
+This release has been tracked through the [0.4.0 project](https://github.com/cloud-hypervisor/cloud-hypervisor/projects/4).
+
+Highlights for `cloud-hypervisor` version 0.4.0 include:
+
+### Dynamic virtual CPUs addition
+
+As a way to vertically scale Cloud-Hypervisor guests, we now support dynamically
+adding virtual CPUs to the guests, a mechanism also known as CPU hot plug.
+Through hardware-reduced ACPI notifications, Cloud Hypervisor can now add CPUs
+to an already running guest and the high level operations for that process are
+documented [here](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/master/docs/hotplug.md)
+
+During the next release cycles we are planning to extend Cloud Hypervisor
+hot plug framework to other resources, namely PCI devices and memory.
+
+### Programmatic firmware tables generation
+
+As part of the CPU hot plug feature enablement, and as a requirement for hot
+plugging other resources like devices or RAM, we added support for
+programmatically generating the needed ACPI tables. Through a dedicated
+`acpi-tables` crate, we now have a flexible and clean way of generating those
+tables based on the VMM device model and topology.
+
+### Filesystem and block devices vhost-user backends
+
+Our objective of running all Cloud Hypervisor paravirtualized I/O to a
+vhost-user based framework is getting closer as we've added Rust based
+implementations for vhost-user-blk and virtiofs backends. Together with the
+vhost-user-net backend that came with the 0.3.0 release, this will form the
+default Cloud Hypervisor I/O architecture.
+
+### Guest pause and resume
+
+As an initial requiremnt for enabling live migration, we added support for
+pausing and resuming any VMM components. As an intermediate step towards live
+migration, the upcoming guest snapshotting feature will be based on the pause
+and resume capabilities.
+
+### Userspace IOAPIC by default
+
+As a way to simplify our device manager implementation, but also in order to
+stay away from privileged rings as often as possible, any virtio devices
```

Comment:
I would say `any device` instead of `any virtio devices`.

Reply:
Fixed. I was influenced by the corresponding PR title, but that indeed applies to any device since then. Neat.